### PR TITLE
Cut out mixer specific keys on config map if user disables mixer

### PR DESCRIPTION
**Pull request by jwendell**

### CODE REVIEW

File: install/kubernetes/helm/istio/templates/configmap.yaml [hunk 1]
```diff
@@ -11,18 +11,22 @@ metadata:
     release: {{ .Release.Name }}
 data:
   mesh: |-
+    {{- if .Values.mixer.policy.enabled }}
     # Set the following variable to true to disable policy checks by the Mixer.
     # Note that metrics will still be reported to the Mixer.
     disablePolicyChecks: {{ .Values.global.disablePolicyChecks }}
+    {{- end }}
 
+    {{- if .Values.mixer.telemetry.enabled }}
```
Comment (douglas-reid) [inline install/kubernetes/helm/istio/templates/configmap.yaml]: i don't think this is correct. it is possible to have tracing without `istio-telemetry` and this flag is for envoys.

Comment (jwendell) [reply install/kubernetes/helm/istio/templates/configmap.yaml]: hmm, that's right. I'll update.

File: install/kubernetes/helm/istio/templates/configmap.yaml [hunk 2]
```diff
@@ -11,18 +11,20 @@ metadata:
...
     # Note that metrics will still be reported to the Mixer.
     disablePolicyChecks: {{ .Values.global.disablePolicyChecks }}
+    {{- end }}
 
     # Set enableTracing to false to disable request tracing.
     enableTracing: {{ .Values.global.enableTracing }}
 
     # Set accessLogFile to empty string to disable access log.
     accessLogFile: "{{ .Values.global.proxy.accessLogFile }}"
+    {{- if or .Values.mixer.policy.enabled .Values.mixer.telemetry.enabled }}
```
Comment (linsun) [inline install/kubernetes/helm/istio/templates/configmap.yaml]: did this miss an end block?

File: install/kubernetes/helm/istio/templates/configmap.yaml [hunk 3]
```diff
@@ -11,18 +11,20 @@ metadata:
     release: {{ .Release.Name }}
 data:
   mesh: |-
+    {{- if .Values.mixer.policy.enabled }}
     # Set the following variable to true to disable policy checks by the Mixer.
     # Note that metrics will still be reported to the Mixer.
     disablePolicyChecks: {{ .Values.global.disablePolicyChecks }}
+    {{- end }}
```
Comment (linsun) [inline install/kubernetes/helm/istio/templates/configmap.yaml]: @douglas-reid is there any value to disable mixer check explicitly when policy is not enabled?

Comment (douglas-reid) [reply install/kubernetes/helm/istio/templates/configmap.yaml]: @linsun do you mind clarifying the question?  are you asking: "if no policies are configured, should calling `istio-policy` be auto-disabled?" If so, then I think that is a desired state -- just not one that is supported *yet*.

Comment (linsun) [reply install/kubernetes/helm/istio/templates/configmap.yaml]: @douglas-reid  Nothing magic behavior here. :).  I was just trying to see if we should configure explicitly `disablePolicyChecks: false` in configmap when policy isn't enabled.  If there is no value, @jwendell 's update LGTM.

File: install/kubernetes/helm/istio/templates/configmap.yaml [hunk 4]
```diff
@@ -11,18 +11,20 @@ metadata:
     release: {{ .Release.Name }}
 data:
   mesh: |-
+    {{- if .Values.mixer.policy.enabled }}
     # Set the following variable to true to disable policy checks by the Mixer.
     # Note that metrics will still be reported to the Mixer.
     disablePolicyChecks: {{ .Values.global.disablePolicyChecks }}
```
Comment (rshriram) [inline install/kubernetes/helm/istio/templates/configmap.yaml]: Remove the disablePolicyChecks thing from the global stuff. Move this into the if block below.
and set disablePolicyChecks : {{ ! .Values.mixer.policy.enabled }} or some such way..

**Pull request closed without merging**